// mongodb-stitch-core-sdk/dist/umd/internal/net/StitchAppRoutes.js
(function (factory) {
    if (typeof module === "object" && typeof module.exports === "object") {
        var v = factory(require, exports);
        if (v !== undefined) module.exports = v;
    }
    else if (typeof define === "function" && define.amd) {
        define(["require", "exports", "../../services/internal/StitchServiceRoutes", "./StitchAppAuthRoutes"], factory);
    }
})(function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var StitchServiceRoutes_1 = require("../../services/internal/StitchServiceRoutes");
    var StitchAppAuthRoutes_1 = require("./StitchAppAuthRoutes");
    var BASE_ROUTE = "/api/client/v2.0";
    exports.BASE_ROUTE = BASE_ROUTE;
    function getAppRoute(clientAppId) {
        return BASE_ROUTE + ("/app/" + clientAppId);
    }
    exports.getAppRoute = getAppRoute;
    function getFunctionCallRoute(clientAppId) {
        return getAppRoute(clientAppId) + "/functions/call";
    }
    exports.getFunctionCallRoute = getFunctionCallRoute;
    var StitchAppRoutes = (function () {
        function StitchAppRoutes(clientAppId) {
            this.clientAppId = clientAppId;
            this.authRoutes = new StitchAppAuthRoutes_1.default(clientAppId);
            this.serviceRoutes = new StitchServiceRoutes_1.default(clientAppId);
            this.functionCallRoute = getFunctionCallRoute(clientAppId);
        }
        return StitchAppRoutes;
    }());
    exports.StitchAppRoutes = StitchAppRoutes;
});
