// mongodb-stitch-core-sdk/dist/umd/auth/providers/serverapikey/ServerApiKeyCredential.js
(function (factory) {
    if (typeof module === "object" && typeof module.exports === "object") {
        var v = factory(require, exports);
        if (v !== undefined) module.exports = v;
    }
    else if (typeof define === "function" && define.amd) {
        define(["require", "exports", "../../ProviderCapabilities", "./ServerApiKeyAuthProvider"], factory);
    }
})(function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var ProviderCapabilities_1 = require("../../ProviderCapabilities");
    var ServerApiKeyAuthProvider_1 = require("./ServerApiKeyAuthProvider");
    var Fields;
    (function (Fields) {
        Fields["KEY"] = "key";
    })(Fields || (Fields = {}));
    var ServerApiKeyCredential = (function () {
        function ServerApiKeyCredential(key, providerName) {
            if (providerName === void 0) { providerName = ServerApiKeyAuthProvider_1.default.DEFAULT_NAME; }
            var _this = this;
            this.providerType = ServerApiKeyAuthProvider_1.default.TYPE;
            this.material = (function () {
                var _a;
                return _a = {}, _a[Fields.KEY] = _this.key, _a;
            })();
            this.providerCapabilities = new ProviderCapabilities_1.default(false);
            this.providerName = providerName;
            this.key = key;
        }
        return ServerApiKeyCredential;
    }());
    exports.default = ServerApiKeyCredential;
});
