// mongodb-stitch-core-sdk/dist/umd/auth/providers/google/GoogleCredential.js
(function (factory) {
    if (typeof module === "object" && typeof module.exports === "object") {
        var v = factory(require, exports);
        if (v !== undefined) module.exports = v;
    }
    else if (typeof define === "function" && define.amd) {
        define(["require", "exports", "../../ProviderCapabilities", "./GoogleAuthProvider"], factory);
    }
})(function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var ProviderCapabilities_1 = require("../../ProviderCapabilities");
    var GoogleAuthProvider_1 = require("./GoogleAuthProvider");
    var Fields;
    (function (Fields) {
        Fields["AUTH_CODE"] = "authCode";
    })(Fields || (Fields = {}));
    var GoogleCredential = (function () {
        function GoogleCredential(authCode, providerName) {
            if (providerName === void 0) { providerName = GoogleAuthProvider_1.default.DEFAULT_NAME; }
            var _this = this;
            this.providerType = GoogleAuthProvider_1.default.TYPE;
            this.material = (function () {
                var _a;
                return _a = {}, _a[Fields.AUTH_CODE] = _this.authCode, _a;
            })();
            this.providerCapabilities = new ProviderCapabilities_1.default(false);
            this.providerName = providerName;
            this.authCode = authCode;
        }
        return GoogleCredential;
    }());
    exports.default = GoogleCredential;
});
